### PR TITLE
Make it easier to create a DataFrame with a single list-column

### LINKED ISSUE #794: Make it easier to create a `DataFrame` with a single list-column
To create a DataFrame with a single list column, we need to wrap the input in `list()` which is annoying:

``` r
library(polars)
options(polars.do_not_repeat_call = TRUE)

pl$DataFrame(list(x = list(1:2, 3:10)))
#> shape: (2, 1)
#> ┌──────────────┐
#> │ x            │
#> │ ---          │
#> │ list[i32]    │
#> ╞══════════════╡
#> │ [1, 2]       │
#> │ [3, 4, … 10] │
#> └──────────────┘

# this should work
pl$DataFrame(x = list(1:2, 3:10))
#> Error: Execution halted with the following contexts
#>    0: In R: in $DataFrame():
#>    1: Encountered the following error in Rust-Polars:
#>          Series length 2 doesn't match the DataFrame height of 8
```
Note that there's no problem to create a DataFrame with >= 2 list-columns:
```r
pl$DataFrame(x = list(1:2, 3:10), y = list(1:2, 3:10))
#> shape: (2, 2)
#> ┌──────────────┬──────────────┐
#> │ x            ┆ y            │
#> │ ---          ┆ ---          │
#> │ list[i32]    ┆ list[i32]    │
#> ╞══════════════╪══════════════╡
#> │ [1, 2]       ┆ [1, 2]       │
#> │ [3, 4, … 10] ┆ [3, 4, … 10] │
#> └──────────────┴──────────────┘
```

### PR DESCRIPTION
**Pull request by etiennebacher**
Close #794

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -40,6 +40,10 @@
         is changed to `$versions$r_package` (#791).
     -   `$rust_polars`, which indicates the version of the dependent Rust Polars,
         is changed to `$versions$rust_crate` (#791).
+-   New behavior when creating a `DataFrame` with a single list-variable. 
+    `pl$DataFrame(x = list(1:2, 3:4))` used to create a `DataFrame` with two
+    columns named "new_column" and "new_column_1", which was unexpected. It now
+    produces a `DataFrame` with a single `list` variable (#794).
```
Comment (eitsupi) [inline NEWS.md:46]: I suppose it affects to `$with_columns` etc. too.
Could you check them?

**Review by eitsupi — CHANGES_REQUESTED** [review]
Good point, thanks!

File: NEWS.md (hunk 1 above)
Comment (etiennebacher) [reply NEWS.md:46]: Thanks I added some tests

Comment (eitsupi) [reply NEWS.md:46]: Thanks, could you please mention in the NEWS the other functions that are affected?

**Review by eitsupi — APPROVED** [review]

File: NEWS.md [hunk 2]
```diff
@@ -40,6 +40,10 @@
         is changed to `$versions$r_package` (#791).
     -   `$rust_polars`, which indicates the version of the dependent Rust Polars,
         is changed to `$versions$rust_crate` (#791).
+-   New behavior when creating a `DataFrame` with a single list-variable.
```
Comment (eitsupi) [inline NEWS.md:43]:
```suggestion
-   New behavior when creating a `DataFrame` with a single list-variable.
```

Comment (etiennebacher) [reply NEWS.md:43]: Can we avoid this type of changes? This is something that `task format-r` doesn't fix

**Comment by etiennebacher** [comment]
Note that behavior for creating list-variable in `$agg()` was already broken so I'll open a separate issue for that

File: NEWS.md (hunk 2 above)
Comment (eitsupi) [reply NEWS.md:43]: An auto-formatter for Markdown must be installed.
VSCode can easily handle this with extensions, but I am not familiar with the tools available for other editors. Perhaps Pretitter is well known?

File: NEWS.md (hunk 1 above)
Comment (etiennebacher) [reply NEWS.md]: done

File: NEWS.md (hunk 2 above)
Comment (etiennebacher) [reply NEWS.md]: I don't know how that works with RStudio

**Pull request merged**